src/psyclone/psyGen.py

```diff
@@ -1761,6 +1764,46 @@ def _rename_psyir(self, suffix):
                     new_declaration,
                     partial_datatype=sym.datatype.partial_datatype)
                 # pylint: enable=protected-access
+            # Or the DataTypeSymbol is a StructureType, in which case we
+            # go through its procedure components.
+            elif isinstance(sym.datatype, StructureType):
```

Comment:
I think psyGen.py is really the only place where this would be used.
Explicitly, what it does is `procedure, nopass :: code => testkern_code` => `procedure, nopass :: code => testkern_new..._code` so it only renames the "target", which is stored as "initial_value" in the component, no the procedure itself.

Reply:
I ended up writing this procedure as codecov would not agree I was testing one of these cases. See `StructureType.replace_procedure_component_initial_value` in my last commit.